docs/migration.md

```diff
@@ -477,11 +477,28 @@ In Pydantic V2, we recognize that the value is an instance of one of the cases a
 
 #### Required, optional, and nullable fields
 
-Pydantic V1 had a somewhat loose idea about "required" versus "nullable" fields. In Pydantic V2, these concepts are
-more clearly defined.
+Pydantic V2 changes some of the logic for specifying whether a field annotated as `Optional` is required
+(i.e., has no default value) or not (i.e., has a default value of `None`), and now more closely matches the
+behavior of `dataclasses`. Similarly, fields annotated as `Any` no longer have a default value of `None`.
 
-Pydantic V2 will move to match `dataclasses`, thus you may explicitly specify a field as `required` or `optional` and whether the field accepts `None` or not.
+The following table describes the behavior of field annotations in V2:
 
+| State                                            | Field Definition           |
+|--------------------------------------------------|----------------------------|
+| Required, cannot be `None`                       | `f1: str`                  |
+| Required, can be `None`                          | `f2: Optional[str]`        |
+| Not required, can be `None`                      | `f3: Optional[str] = None` |
```

Comment:
I feel like this could give a slightly incorrect impression that exactly `= None` is needed to transform a required field into a not required one.

I think that it is worth mentioning here that any default value if provided makes a field not required.

Something like this maybe
```suggestion
| Not required, cannot be `None`, is `'abc'` by default | `f3: str = 'abc'`      |
| Required, can be `None`                               | `f2: Optional[str]`      |
| Not required, can be `None`, is `None` by default     | `f3: Optional[str] = None`  |
| Not required, can be `None`, is `'abc'` by default    | `f3: Optional[str] = 'abc'` |
```